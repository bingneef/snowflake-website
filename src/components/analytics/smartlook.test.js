import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Smartlook, { sendEvent } from "./smartlook";

describe("Smartlook", () => {
  describe("env = development", () => {
    beforeEach(() => {
      process.env.NODE_ENV = "development";
    });

    it("renders without error", () => {
      const div = document.createElement("div");

      ReactDOM.render(<Smartlook />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it("matches the snapshot", () => {
      const component = renderer.create(<Smartlook />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("env = production", () => {
    beforeEach(() => {
      process.env.NODE_ENV = "production";
    });

    it("renders without error", () => {
      const div = document.createElement("div");

      ReactDOM.render(<Smartlook />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it("matches the snapshot", () => {
      const component = renderer.create(<Smartlook />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("#sendEvent", () => {
  it("logs if smartlook is not defined", () => {
    console.log = jest.fn();
    window.smartlook = undefined;

    sendEvent("TestEvent");

    expect(console.log).toHaveBeenCalledWith(
      "Not sending event: TestEvent, not in production environment"
    );
  });

  it("sends to smartlook if defined", () => {
    window.smartlook = jest.fn();

    sendEvent("TestEvent", { a: true });

    expect(window.smartlook).toHaveBeenCalledWith("track", "TestEvent", {
      a: true
    });
  });
});
