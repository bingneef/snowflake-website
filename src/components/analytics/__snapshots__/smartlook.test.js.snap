// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smartlook env = development matches the snapshot 1`] = `null`;

exports[`Smartlook env = production matches the snapshot 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "
          window.smartlook||(function(d) {
          var o=smartlook=function(){ o.api.push(arguments)},h=d.getElementsByTagName('head')[0];
          var c=d.createElement('script');o.api=new Array();c.async=true;c.type='text/javascript';
          c.charset='utf-8';c.src='https://rec.smartlook.com/recorder.js';h.appendChild(c);
          })(document);
          smartlook('init', '1918e07593de805de944505d6e91fb8ceec2b696');
          ",
    }
  }
/>
`;
